components/empty-state/empty-state-illustrated-button.js

```diff
@@ -1,63 +1,139 @@
 import { RtlMixin } from '../../mixins/rtl-mixin.js';
 import { runAsync } from '../../directives/run-async/run-async.js';
+import { styleMap } from 'lit/directives/style-map.js';
+import { unsafeSVG } from 'lit/directives/unsafe-svg.js';
 
 /**
  * The `d2l-empty-state-illustrated-button` component is an empty state component that displays an illustration and action button. The illustration property can be set to use one of the preset illustrations or a custom SVG illustration can be added in the default slot.
  * @fires d2l-empty-state-action - Dispatched when the action button is clicked
- * @slot - Custom SVG content if illustration property is not set
+ * @slot - Custom SVG content if `illustration-name` property is not set
  */
 class EmptyStateIllustratedButton extends RtlMixin(LitElement) {
 
 	static get properties() {
 		return {
 			/**
 			 * Optional: The action text to be used in the subtle button
-			 * @type {String}
+			 * @type {string}
 			 */
 			actionText: { type: String, attribute: 'action-text' },
 			/**
 			 * REQUIRED: A description giving details about the empty state
-			 * @type {String}
+			 * @type {string}
 			 */
 			description: { type: String },
 			/**
 			 * Optional: The name of the preset image you would like to display in the component
-			 * @type {String}
+			 * @type {string}
 			 */
-			illustration: { type: String },
+			illustrationName: { type: String, attribute: 'illustration-name' },
+			/**
+			 * Optional: This will change the action button to use a primary button instead of the default subtle button
+			 * @type {boolean}
+			 */
+			primary: { type: Boolean },
 			/**
 			 * REQUIRED: A title for the empty state
-			 * @type {String}
+			 * @type {string}
 			 */
-			title: { type: String }
+			titleText: { type: String, attribute: 'title-text' },
+			_contentWidth: { type: Number }
 		};
 	}
 
 	static get styles() {
-		return [emptyStateStyles, emptyStateIllustratedStyles, bodyCompactStyles, heading2Styles];
+		return [bodyCompactStyles, emptyStateStyles, emptyStateIllustratedStyles];
+	}
+
+	constructor() {
+		super();
+
+		this._contentWidth = 700;
+		this._resizeObserver = new ResizeObserver(this._onResize.bind(this));
+	}
+
+	connectedCallback() {
+		super.connectedCallback();
+		this._resizeObserver.observe(this);
+	}
+
+	disconnectedCallback() {
+		super.disconnectedCallback();
+		this._resizeObserver.disconnect();
+	}
+
+	firstUpdated(changedProperties) {
+		super.firstUpdated(changedProperties);
+
+		if (!this.titleText || this.titleText.length === 0) {
+			console.warn('d2l-empty-state-illustrated-button component requires titleText.');
+		}
+
+		if (!this.description || this.description.length === 0) {
+			console.warn('d2l-empty-state-illustrated-button component requires a description.');
+		}
 	}
 
 	render() {
+		const illustrationContainerStyle = {
+			height: `${Math.min(this._contentWidth, 500) / 1.5}px`,
+		};
+
+		const titleSmall = this._contentWidth <= 615;
```

Comment:
The way this is written, every time `_contentWidth` changes it's going to cause a re-render. But in reality, only when the value drops below `615` or `500` do we really need to re-render.

I think you can optimize it a bit by creating `_titleSmall` boolean and `_illustrationHeight` state properties that get set from the resize handler. Then only when those values change will Lit actually re-render.

Reply:
Also I think you only care about `_illustrationHeight` (and `illustrationContainerStyle`) if `illustration-name` is actually set -- so again some re-renders could be avoided I think when they're using a custom SVG.

Reply:
I did originally do it that way with a `_titleSmall` attribute. The reason I changed it was so that I could preset the height of the container before the illustration loads in to prevent the initial jitter. I could optimize it somewhat to not update the height when width is `> 500` since height no longer changes but if it is less than that, then it would need to have the exact width measurements in order to get the correct height. Is that what you are proposing?

Agree with you on the second comment, it can be made more efficient there. Thanks

Reply:
Yeah exactly. I was proposing two separate state properties -- one to make the title small and one for the illustration height (that would only get changed when width drops below 500), but you're correct that you could accomplish the same thing by just not updating `_contentWidth` until it drops below 500 (or goes back above 500).